lib/eve.js

```diff
@@ -0,0 +1,63 @@
+import { useState, useEffect } from 'preact/hooks'
+
+// throttle a function call to avoid calling it too much
+export const throttle = (fn, delay, t, call = () => ((t = 0), fn())) => () =>
+  t || (t = setTimeout(call, delay))
+
+// create an observable or event
+export const eve = (...args) => {
+  const subs = new Set()
+  const sub = (fn) => (subs.add(fn), () => subs.delete(fn))
+  const next = () => new Promise(once)
+  const once = (fn) =>
+    subs.add(function $(value, prev) {
+      subs.delete($)
+      fn(value, prev)
+    })
```

Comment:
`once` will add a function that will be triggered only once.
For that I create a wrapper function `$` *(name just used for reference)*
and then delete it from the subs set as soon as it's called.

`prev` is the previous value but this isn't applicable for events, it exists as this method is shared between event and observables.
More on that later

Reply:
`next` is just the promisified version of that, it takes no arguments and will wait for the first value returned